modules/nuxeo-apidoc-core/src/main/java/org/nuxeo/apidoc/snapshot/VersionComparator.java

```diff
@@ -0,0 +1,110 @@
+/*
+ * (C) Copyright 2006-2020 Nuxeo SA (http://nuxeo.com/) and others.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * Contributors:
+ *     Thierry Delprat
+ *     Anahide Tchertchian
+ */
+package org.nuxeo.apidoc.snapshot;
+
+import java.util.Comparator;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+/**
+ * Comparator for versions using 1 to 3 digits.
+ * <p>
+ * Makes sure version "11.2.3" is < "11.2.10" (so simple String ordering cannot be used).
+ *
+ * @since 20.0.0
+ */
+public class VersionComparator implements Comparator<String> {
+
+    protected static final Pattern VERSION_REGEX = Pattern.compile(
+            "^(\\d+)(?:\\.(\\d+))?(?:\\.(\\d+))?(?:-(\\D+))?(?:(\\d+))?$", Pattern.CASE_INSENSITIVE);
+
+    public static boolean isVersion(String candidate) {
+        return candidate != null && candidate.matches(VersionComparator.VERSION_REGEX.toString());
```

Comment:
Should be `&& VERSION_REGEX.matcher(candidate).matches()` to avoid recompiling the regexp each time.